x-pack/plugins/ml/server/plugin.ts

```diff
@@ -212,6 +217,12 @@ export class MlServerPlugin
       });
     }
 
+    if (plugins.usageCollection) {
+      this.kibanaIndexConfig.subscribe((config) => {
+        registerCollector(plugins.usageCollection!, config.kibana.index);
```

Comment:
`!` is not requred since you have a type guard via the if statement above

```suggestion
        registerCollector(plugins.usageCollection, config.kibana.index);
```